README.md

```diff
@@ -977,7 +978,41 @@ Some of these functions makes an in memory copy of the output. For example: cycl
 Also with the iterator protocol you can create infinite iterables (repeat, cycle, count etc.). These iterables can't be used by all generators. For example combinatory generators require finite iterables.
 Some of the obvious things you can do with arrays, are not efficients with iterables. For example: sorting, shuffling and in general all operations that rely on having the full array at your disposal. In that case the way to go is to convert the iterable in an array and use that.
 
+## when
+when is a helper you can use with es6 spread syntax (the `...` operator). In particular it helps avoid an unnecessarily difficult to read pattern that frequently causes code formatters (prettier, specifically) to emit an undesireable number of lines:
+```js
+const always = true;
+const sometimes = Math.random() > .5;
+
+const arr = {
+  always,
+  ...(
+    sometimes ? [ sometimes ] : []
+  )
+} // [true, true] OR [true]
+```
+
+Instead, you can use the when method like so:
+```js
+const whenArr = {
+  always,
+    ...when(sometimes, [ sometimes ]),
+  ) // [true, true] OR [true]
```

Comment:
I think you should remove this closing brace

Reply:
And the outer braces should be `{` not `[`.